File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.go
func topKFrequent(nums []int, k int) []int {
    freq := make(map[int]int)

    for _, num := range nums{
        freq[num]++
    }

    bucket := make([][]int, len(nums) + 1)
    for num, f := range freq {
        bucket[f] = append(bucket[f], num)
    }

    res := []int{}

    for i:=len(nums);i>=0;i--{
        for _, num := range bucket[i]{
            res = append(res, num)
            if len(res) == k{
                return res
            }
        }
    }
    return res
}

File: Data Structures & Algorithms/remove-element/submission-2.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        vector<int> res;
        for(int n: nums){
            if(n != val){
                res.push_back(n);
            }
        }

        for(int i = 0; i< res.size(); i++){
            nums[i] = res[i];
        }

        return res.size();        
    }
};